enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -378,15 +384,37 @@ class AdjointGenerator
     }
 
     if (isfloat) {
-      IRBuilder<> Builder2(parent);
-      getReverseBuilder(Builder2);
-      auto prediff = diffe(&I, Builder2);
-      setDiffe(&I, Constant::getNullValue(type), Builder2);
 
-      if (!gutils->isConstantValue(I.getOperand(0))) {
-        ((DiffeGradientUtils *)gutils)
-            ->addToInvertedPtrDiffe(I.getOperand(0), prediff, Builder2,
-                                    alignment, OrigOffset);
+      switch (Mode) {
+      case DerivativeMode::ForwardMode: {
+        IRBuilder<> Builder2(&I);
+        getForwardBuilder(Builder2);
+
+        auto diff = Builder2.CreateLoad(
+            gutils->invertPointerM(I.getOperand(0), Builder2));
```

Comment:
I'd move this invert pointer load inot the isconstantvalue check, since only needed in that case.

Reply:
makes sense 😄